File: CSharpCodeBase/aiactions/scaredclass.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class scaredclass :aisequencenode{
 var AI_ScaredClass = Class({__includes=AISequenceNode, init = public void (self, entity){
       var childNodes = {
         AI_GetTargetClass(entity),;
         AI_RunawayFromTarget(entity),;
       }
       AISequenceNode.init(self, entity, childNodes);
 }})
 return AI_ScaredClass}}

File: CSharpCodeBase/aiactions/scaredclass.lua
local Class = require 'utils.hump.class'
local AI_ConditionalNode = require 'utils.ailib.aiifnode'
local AISequenceNode = require 'utils.ailib.aisequence'
local AI_GetTargetClass = require 'aiactions.ai_gettargetclass'
local AI_RunawayFromTarget = require 'aiactions.ai_runawayfromtarget'

local AI_ScaredClass = Class({__includes=AISequenceNode, init = function (self, entity)
      local childNodes = {
        AI_GetTargetClass(entity),
        AI_RunawayFromTarget(entity),
      }
      AISequenceNode.init(self, entity, childNodes)
end})

return AI_ScaredClass